File: hackathon_CPP_SDK/src-gen/nevonex/isopgn/ISOPGN.cpp
/*
 * nevonex/isopgn/ISOPGN.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "ISOPGN.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/isopgn/IsopgnPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

/*PROTECTED REGION ID(ISOPGN.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::isopgn;
using namespace ::nevonex::log;

// Default constructor
ISOPGN::ISOPGN() : m_accpedPos1(0), m_ambAirt(0), m_brkPedPos(0), m_engFuRatePerTime(0), m_engSpd(0), m_engTotOperHrs(0), m_fuLvlPer(0), m_fuLvlPer2(0), m_gndBasdMacSpd(0), m_tachoVehSpd(0), m_totalVehDist(0), m_totVehHrs(0), m_whlBasdMacSpd(0)
{
    // Machine does NOT call _initialize() — Provider manages initialization
    m_interfaceDetailMap = {
    };

    /*PROTECTED REGION ID(ISOPGN__ISOPGN) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/
}

ISOPGN::~ISOPGN()
{
    /*PROTECTED REGION ID(ISOPGN__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/
}

// References
::ecore::EFloat ISOPGN::getAccpedPos1() const
{
    return m_accpedPos1;
}

void ISOPGN::setAccpedPos1(::ecore::EFloat _accpedPos1)
{
    ::ecore::EFloat _old_accpedPos1 = m_accpedPos1;
    m_accpedPos1 = _accpedPos1;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"accpedPos1\" has changed.";
    notifyPropertyChange("accpedPos1", _old_accpedPos1, m_accpedPos1);
}

::ecore::EFloat ISOPGN::getAmbAirt() const
{
    return m_ambAirt;
}

void ISOPGN::setAmbAirt(::ecore::EFloat _ambAirt)
{
    ::ecore::EFloat _old_ambAirt = m_ambAirt;
    m_ambAirt = _ambAirt;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"ambAirt\" has changed.";
    notifyPropertyChange("ambAirt", _old_ambAirt, m_ambAirt);
}

::ecore::EFloat ISOPGN::getBrkPedPos() const
{
    return m_brkPedPos;
}

void ISOPGN::setBrkPedPos(::ecore::EFloat _brkPedPos)
{
    ::ecore::EFloat _old_brkPedPos = m_brkPedPos;
    m_brkPedPos = _brkPedPos;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"brkPedPos\" has changed.";
    notifyPropertyChange("brkPedPos", _old_brkPedPos, m_brkPedPos);
}

::ecore::EFloat ISOPGN::getEngFuRatePerTime() const
{
    return m_engFuRatePerTime;
}

void ISOPGN::setEngFuRatePerTime(::ecore::EFloat _engFuRatePerTime)
{
    ::ecore::EFloat _old_engFuRatePerTime = m_engFuRatePerTime;
    m_engFuRatePerTime = _engFuRatePerTime;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"engFuRatePerTime\" has changed.";
    notifyPropertyChange("engFuRatePerTime", _old_engFuRatePerTime, m_engFuRatePerTime);
}

::ecore::EFloat ISOPGN::getEngSpd() const
{
    return m_engSpd;
}

void ISOPGN::setEngSpd(::ecore::EFloat _engSpd)
{
    ::ecore::EFloat _old_engSpd = m_engSpd;
    m_engSpd = _engSpd;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"engSpd\" has changed.";
    notifyPropertyChange("engSpd", _old_engSpd, m_engSpd);
}

::ecore::EFloat ISOPGN::getEngTotOperHrs() const
{
    return m_engTotOperHrs;
}

void ISOPGN::setEngTotOperHrs(::ecore::EFloat _engTotOperHrs)
{
    ::ecore::EFloat _old_engTotOperHrs = m_engTotOperHrs;
    m_engTotOperHrs = _engTotOperHrs;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"engTotOperHrs\" has changed.";
    notifyPropertyChange("engTotOperHrs", _old_engTotOperHrs, m_engTotOperHrs);
}

::ecore::EFloat ISOPGN::getFuLvlPer() const
{
    return m_fuLvlPer;
}

void ISOPGN::setFuLvlPer(::ecore::EFloat _fuLvlPer)
{
    ::ecore::EFloat _old_fuLvlPer = m_fuLvlPer;
    m_fuLvlPer = _fuLvlPer;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"fuLvlPer\" has changed.";
    notifyPropertyChange("fuLvlPer", _old_fuLvlPer, m_fuLvlPer);
}

::ecore::EFloat ISOPGN::getFuLvlPer2() const
{
    return m_fuLvlPer2;
}

void ISOPGN::setFuLvlPer2(::ecore::EFloat _fuLvlPer2)
{
    ::ecore::EFloat _old_fuLvlPer2 = m_fuLvlPer2;
    m_fuLvlPer2 = _fuLvlPer2;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"fuLvlPer2\" has changed.";
    notifyPropertyChange("fuLvlPer2", _old_fuLvlPer2, m_fuLvlPer2);
}

::ecore::EFloat ISOPGN::getGndBasdMacSpd() const
{
    return m_gndBasdMacSpd;
}

void ISOPGN::setGndBasdMacSpd(::ecore::EFloat _gndBasdMacSpd)
{
    ::ecore::EFloat _old_gndBasdMacSpd = m_gndBasdMacSpd;
    m_gndBasdMacSpd = _gndBasdMacSpd;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"gndBasdMacSpd\" has changed.";
    notifyPropertyChange("gndBasdMacSpd", _old_gndBasdMacSpd, m_gndBasdMacSpd);
}

::ecore::EFloat ISOPGN::getTachoVehSpd() const
{
    return m_tachoVehSpd;
}

void ISOPGN::setTachoVehSpd(::ecore::EFloat _tachoVehSpd)
{
    ::ecore::EFloat _old_tachoVehSpd = m_tachoVehSpd;
    m_tachoVehSpd = _tachoVehSpd;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"tachoVehSpd\" has changed.";
    notifyPropertyChange("tachoVehSpd", _old_tachoVehSpd, m_tachoVehSpd);
}

::ecore::EString ISOPGN::getTotalVehDist() const
{
    return m_totalVehDist;
}

void ISOPGN::setTotalVehDist(::ecore::EString _totalVehDist)
{
    ::ecore::EString _old_totalVehDist = m_totalVehDist;
    m_totalVehDist = _totalVehDist;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"totalVehDist\" has changed.";
    notifyPropertyChange("totalVehDist", _old_totalVehDist, m_totalVehDist);
}

::ecore::EFloat ISOPGN::getTotVehHrs() const
{
    return m_totVehHrs;
}

void ISOPGN::setTotVehHrs(::ecore::EFloat _totVehHrs)
{
    ::ecore::EFloat _old_totVehHrs = m_totVehHrs;
    m_totVehHrs = _totVehHrs;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"totVehHrs\" has changed.";
    notifyPropertyChange("totVehHrs", _old_totVehHrs, m_totVehHrs);
}

::ecore::EFloat ISOPGN::getWhlBasdMacSpd() const
{
    return m_whlBasdMacSpd;
}

void ISOPGN::setWhlBasdMacSpd(::ecore::EFloat _whlBasdMacSpd)
{
    ::ecore::EFloat _old_whlBasdMacSpd = m_whlBasdMacSpd;
    m_whlBasdMacSpd = _whlBasdMacSpd;
    SDK_FCAL_LOG(SeverityLevel::debug) << "ISOPGN property \"whlBasdMacSpd\" has changed.";
    notifyPropertyChange("whlBasdMacSpd", _old_whlBasdMacSpd, m_whlBasdMacSpd);
}

// FCAL timestamp/valid
::ecore::ELong ISOPGN::getAccpedPos1_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isAccpedPos1_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getAmbAirt_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isAmbAirt_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getBrkPedPos_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isBrkPedPos_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getEngFuRatePerTime_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isEngFuRatePerTime_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getEngSpd_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isEngSpd_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getEngTotOperHrs_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isEngTotOperHrs_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getFuLvlPer_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isFuLvlPer_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getFuLvlPer2_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isFuLvlPer2_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getGndBasdMacSpd_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isGndBasdMacSpd_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getTachoVehSpd_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isTachoVehSpd_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getTotalVehDist_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isTotalVehDist_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getTotVehHrs_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isTotVehHrs_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

::ecore::ELong ISOPGN::getWhlBasdMacSpd_Timestamp()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).getTimestamp();
    return 0;
}

::ecore::EBoolean ISOPGN::isWhlBasdMacSpd_Valid()
{
    if (m_interfaceDetailMap.count(0))
        return m_interfaceDetailMap.at(0).isValid();
    return false;
}

// PropertyChange delegation (diamond inheritance requires explicit override)
void ISOPGN::addPropertyChangeListener(
        ::nevonex::types::PropertyChangeListener _listener)
{
    using namespace ::nevonex::types;
    PropertyChange::addPropertyChangeListener(_listener);
}

void ISOPGN::removePropertyChangeListener(
        ::nevonex::types::PropertyChangeListener _listener)
{
    using namespace ::nevonex::types;
    PropertyChange::removePropertyChangeListener(_listener);
}

void ISOPGN::notifyPropertyChange(::ecore::EString const &_name,
        ::ecore::EJavaObject const &_oldValue,
        ::ecore::EJavaObject const &_newValue)
{
    using namespace ::nevonex::types;
    PropertyChange::notifyPropertyChange(_name, _oldValue, _newValue);
}

File: com.bosch.fsp.hackathon.gen/src/com/bosch/nevonex/isopgn/impl/ISOPGN.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.isopgn.impl;

import com.bosch.fsp.logger.FCALLogs;
import com.bosch.fsp.logger.LoggerConstants;

import com.bosch.fsp.platform.InterfaceDetails;

import com.bosch.fsp.runtime.feature.IMachine;

import com.bosch.nevonex.common.ITopicObject;
import com.bosch.nevonex.common.impl.TopicObject;

import com.bosch.nevonex.isopgn.IISOPGN;

import com.bosch.nevonex.types.IPropertyChange;
import com.bosch.nevonex.types.impl.TypesPackage;

import java.beans.PropertyChangeEvent;
import java.beans.PropertyChangeListener;

import java.lang.reflect.InvocationTargetException;

import java.util.Collection;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.eclipse.emf.common.util.BasicEList;
import org.eclipse.emf.common.util.EList;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.util.EDataTypeUniqueEList;

/**
 * An implementation of the model object '<em><b>ISOPGN</b></em>'.
 * @generated
 */
public class ISOPGN extends TopicObject implements IISOPGN {
	protected EList<PropertyChangeListener> listeners;
	protected Map<String, InterfaceDetails> interfaceDetailMap = new HashMap<>();
	protected float accpedPos1 = 0.0f;
	protected float ambAirt = 0.0f;
	protected float brkPedPos = 0.0f;
	protected float engFuRatePerTime = 0.0f;
	protected float engSpd = 0.0f;
	protected float engTotOperHrs = 0.0f;
	protected float fuLvlPer = 0.0f;
	protected float fuLvlPer2 = 0.0f;
	protected float gndBasdMacSpd = 0.0f;
	protected float tachoVehSpd = 0.0f;
	protected String totalVehDist = null;
	protected float totVehHrs = 0.0f;
	protected float whlBasdMacSpd = 0.0f;

	/**
	 * @generated
	 */
	protected ISOPGN() {
		super();
		initInterfaceDetails();
	}

	private void initInterfaceDetails() {
		// Interface details initialization
	}

	/**
	 * @generated
	 */
	@Override
	protected EClass eStaticClass() {
		return IsopgnPackage.Literals.ISOPGN;
	}

	public float getAccpedPos1() {
		return accpedPos1;
	}

	public void setAccpedPos1(float newAccpedPos1) {
		accpedPos1 = newAccpedPos1;
	}

	public float getAmbAirt() {
		return ambAirt;
	}

	public void setAmbAirt(float newAmbAirt) {
		ambAirt = newAmbAirt;
	}

	public float getBrkPedPos() {
		return brkPedPos;
	}

	public void setBrkPedPos(float newBrkPedPos) {
		brkPedPos = newBrkPedPos;
	}

	public float getEngFuRatePerTime() {
		return engFuRatePerTime;
	}

	public void setEngFuRatePerTime(float newEngFuRatePerTime) {
		engFuRatePerTime = newEngFuRatePerTime;
	}

	public float getEngSpd() {
		return engSpd;
	}

	public void setEngSpd(float newEngSpd) {
		engSpd = newEngSpd;
	}

	public float getEngTotOperHrs() {
		return engTotOperHrs;
	}

	public void setEngTotOperHrs(float newEngTotOperHrs) {
		engTotOperHrs = newEngTotOperHrs;
	}

	public float getFuLvlPer() {
		return fuLvlPer;
	}

	public void setFuLvlPer(float newFuLvlPer) {
		fuLvlPer = newFuLvlPer;
	}

	public float getFuLvlPer2() {
		return fuLvlPer2;
	}

	public void setFuLvlPer2(float newFuLvlPer2) {
		fuLvlPer2 = newFuLvlPer2;
	}

	public float getGndBasdMacSpd() {
		return gndBasdMacSpd;
	}

	public void setGndBasdMacSpd(float newGndBasdMacSpd) {
		gndBasdMacSpd = newGndBasdMacSpd;
	}

	public float getTachoVehSpd() {
		return tachoVehSpd;
	}

	public void setTachoVehSpd(float newTachoVehSpd) {
		tachoVehSpd = newTachoVehSpd;
	}

	public String getTotalVehDist() {
		return totalVehDist;
	}

	public void setTotalVehDist(String newTotalVehDist) {
		totalVehDist = newTotalVehDist;
	}

	public float getTotVehHrs() {
		return totVehHrs;
	}

	public void setTotVehHrs(float newTotVehHrs) {
		totVehHrs = newTotVehHrs;
	}

	public float getWhlBasdMacSpd() {
		return whlBasdMacSpd;
	}

	public void setWhlBasdMacSpd(float newWhlBasdMacSpd) {
		whlBasdMacSpd = newWhlBasdMacSpd;
	}

	/**
	 * @generated
	 */
	public void addPropertyChangeListener(PropertyChangeListener listener) {
		if (listeners == null) {
			listeners = new BasicEList<>();
		}
		listeners.add(listener);
	}

	/**
	 * @generated
	 */
	public void removePropertyChangeListener(PropertyChangeListener listener) {
		if (listeners != null) {
			listeners.remove(listener);
		}
	}

	/**
	 * @generated
	 */
	public void notifyPropertyChange(String name, Object oldValue, Object newValue) {
		if (listeners != null) {
			for (PropertyChangeListener listener : this.listeners) {
				listener.propertyChange(new PropertyChangeEvent(this, name, oldValue, newValue));
			}
		}
	}

	/**
	 * @generated
	 */
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case IsopgnPackage.IISOPGN__ACCPED_POS1:
			return getAccpedPos1();
		case IsopgnPackage.IISOPGN__AMB_AIRT:
			return getAmbAirt();
		case IsopgnPackage.IISOPGN__BRK_PED_POS:
			return getBrkPedPos();
		case IsopgnPackage.IISOPGN__ENG_FU_RATE_PER_TIME:
			return getEngFuRatePerTime();
		case IsopgnPackage.IISOPGN__ENG_SPD:
			return getEngSpd();
		case IsopgnPackage.IISOPGN__ENG_TOT_OPER_HRS:
			return getEngTotOperHrs();
		case IsopgnPackage.IISOPGN__FU_LVL_PER:
			return getFuLvlPer();
		case IsopgnPackage.IISOPGN__FU_LVL_PER2:
			return getFuLvlPer2();
		case IsopgnPackage.IISOPGN__GND_BASD_MAC_SPD:
			return getGndBasdMacSpd();
		case IsopgnPackage.IISOPGN__TACHO_VEH_SPD:
			return getTachoVehSpd();
		case IsopgnPackage.IISOPGN__TOTAL_VEH_DIST:
			return getTotalVehDist();
		case IsopgnPackage.IISOPGN__TOT_VEH_HRS:
			return getTotVehHrs();
		case IsopgnPackage.IISOPGN__WHL_BASD_MAC_SPD:
			return getWhlBasdMacSpd();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	/**
	 * @generated
	 */
	@SuppressWarnings("unchecked")
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case IsopgnPackage.IISOPGN__ACCPED_POS1:
			setAccpedPos1((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__AMB_AIRT:
			setAmbAirt((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__BRK_PED_POS:
			setBrkPedPos((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__ENG_FU_RATE_PER_TIME:
			setEngFuRatePerTime((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__ENG_SPD:
			setEngSpd((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__ENG_TOT_OPER_HRS:
			setEngTotOperHrs((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__FU_LVL_PER:
			setFuLvlPer((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__FU_LVL_PER2:
			setFuLvlPer2((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__GND_BASD_MAC_SPD:
			setGndBasdMacSpd((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__TACHO_VEH_SPD:
			setTachoVehSpd((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__TOTAL_VEH_DIST:
			setTotalVehDist((String) newValue);
			return;
		case IsopgnPackage.IISOPGN__TOT_VEH_HRS:
			setTotVehHrs((Float) newValue);
			return;
		case IsopgnPackage.IISOPGN__WHL_BASD_MAC_SPD:
			setWhlBasdMacSpd((Float) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	/**
	 * @generated
	 */
	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case IsopgnPackage.IISOPGN__ACCPED_POS1:
			setAccpedPos1(0.0f);
			return;
		case IsopgnPackage.IISOPGN__AMB_AIRT:
			setAmbAirt(0.0f);
			return;
		case IsopgnPackage.IISOPGN__BRK_PED_POS:
			setBrkPedPos(0.0f);
			return;
		case IsopgnPackage.IISOPGN__ENG_FU_RATE_PER_TIME:
			setEngFuRatePerTime(0.0f);
			return;
		case IsopgnPackage.IISOPGN__ENG_SPD:
			setEngSpd(0.0f);
			return;
		case IsopgnPackage.IISOPGN__ENG_TOT_OPER_HRS:
			setEngTotOperHrs(0.0f);
			return;
		case IsopgnPackage.IISOPGN__FU_LVL_PER:
			setFuLvlPer(0.0f);
			return;
		case IsopgnPackage.IISOPGN__FU_LVL_PER2:
			setFuLvlPer2(0.0f);
			return;
		case IsopgnPackage.IISOPGN__GND_BASD_MAC_SPD:
			setGndBasdMacSpd(0.0f);
			return;
		case IsopgnPackage.IISOPGN__TACHO_VEH_SPD:
			setTachoVehSpd(0.0f);
			return;
		case IsopgnPackage.IISOPGN__TOTAL_VEH_DIST:
			setTotalVehDist(null);
			return;
		case IsopgnPackage.IISOPGN__TOT_VEH_HRS:
			setTotVehHrs(0.0f);
			return;
		case IsopgnPackage.IISOPGN__WHL_BASD_MAC_SPD:
			setWhlBasdMacSpd(0.0f);
			return;
		}
		super.eUnset(featureID);
	}

	/**
	 * @generated
	 */
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case IsopgnPackage.IISOPGN__ACCPED_POS1:
			return accpedPos1 != 0.0f;
		case IsopgnPackage.IISOPGN__AMB_AIRT:
			return ambAirt != 0.0f;
		case IsopgnPackage.IISOPGN__BRK_PED_POS:
			return brkPedPos != 0.0f;
		case IsopgnPackage.IISOPGN__ENG_FU_RATE_PER_TIME:
			return engFuRatePerTime != 0.0f;
		case IsopgnPackage.IISOPGN__ENG_SPD:
			return engSpd != 0.0f;
		case IsopgnPackage.IISOPGN__ENG_TOT_OPER_HRS:
			return engTotOperHrs != 0.0f;
		case IsopgnPackage.IISOPGN__FU_LVL_PER:
			return fuLvlPer != 0.0f;
		case IsopgnPackage.IISOPGN__FU_LVL_PER2:
			return fuLvlPer2 != 0.0f;
		case IsopgnPackage.IISOPGN__GND_BASD_MAC_SPD:
			return gndBasdMacSpd != 0.0f;
		case IsopgnPackage.IISOPGN__TACHO_VEH_SPD:
			return tachoVehSpd != 0.0f;
		case IsopgnPackage.IISOPGN__TOTAL_VEH_DIST:
			return totalVehDist != null;
		case IsopgnPackage.IISOPGN__TOT_VEH_HRS:
			return totVehHrs != 0.0f;
		case IsopgnPackage.IISOPGN__WHL_BASD_MAC_SPD:
			return whlBasdMacSpd != 0.0f;
		}
		return super.eIsSet(featureID);
	}
} //ISOPGN
